### PR TITLE
Add flag for specifying config path location

### LINKED ISSUE #329: Specify config path
Since we can make more specific configs based off of spago.dhall, we should be able to specify specific configs.

Example:

MunProjekti420 has the following files:

```
src/Main.purs
src/Generated.purs
test/Main.purs
```

I would like to build and run only src/Main.purs, but non ni, Generated and Test.Main ovat rikki.

Jesusteippilla:

```dhall
# kekkonen420.dhall
./spago.dhall // { sources = [ "src/Main.purs" ] }
```

Kaunis. Nyt voimme rakentaa projektimme rauhassa.

```
spago run -c kekkonen420.dhall
```

### PR DESCRIPTION
**Pull request by justinwoo**
### Description of the change

fixes #329, specify config path globally if desired for specific usages

e.g.

`spago.dhall`

```dhall
-- ...
, sources =
    [ "codegen/**/*.purs", "src/**/*.purs", "test/**/*.purs" ]
}
```

i don't want to build with all the sources.

`codegen.dhall`

```dhall
./spago.dhall // {
  sources = [
	"src/**/*.purs"
  ]
}
```

```
$ spago -x codegen.dhall build
# works fine
```

other uses:

"optional app separate from library"

"test config overrides and modifications"

"some other config modifications"

### Checklist:

- [ ] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

**Comment by justinwoo** [comment]
> /bin/sh: psa: command not found

in osx build

**Comment by f-f** [comment]
@justinwoo that error there is because of the test harness and is a normal (I'll refactor the tests so it doesn't happen), but the tests fail because `spago update-set` is flaky on Travis machine (I'll put a PR up for this)

I'm working through my backlog, I'll review this later today

File: src/Spago/Prelude.hs [hunk 1]
```diff
@@ -134,9 +135,10 @@ instance Show SpagoError where
 data UsePsa = UsePsa | NoPsa
 
 data GlobalOptions = GlobalOptions
-  { globalDebug  :: Bool
-  , globalUsePsa :: UsePsa
-  , globalJobs   :: Int
+  { globalDebug      :: Bool
+  , globalUsePsa     :: UsePsa
+  , globalJobs       :: Int
+  , globalConfigPath :: Maybe Text
```
[the remark this answers is not in the document]
Comment (justinwoo) [reply src/Spago/Prelude.hs]: let's try it

**Comment by justinwoo** [comment]
I think it would also be good to just disallow automatic editing for anything that doesn't conform to the expected Expr structure, and instead just print warnings on what the users should do manually with their frankenstein-ass configs

**Comment by f-f** [comment]
@justinwoo that is the current behaviour: if stuff doesn't conform to the expected structure then the editing is not performed. However we do not have the infrastructure to detect when this is the case and print warnings

TL;DR: I believe if you add the tests I mentioned they will just pass

**Comment by f-f** [comment]
This should be good to go. @justinwoo can you take a look?
